pkgs/applications/editors/010-editor/default.nix

```diff
@@ -0,0 +1,83 @@
+{ lib, stdenv, fetchFromGitHub, fetchurl, makeDesktopItem, autoPatchelfHook, copyDesktopItems, perlPackages, zlib, fontconfig, freetype, libxkbcommon, cups, ... }:
+
+let
+  extract = fetchFromGitHub {
+    owner = "lod";
+    repo = "unpack-install-jammer";
+    rev = "41faa5a8c7925db4ff7ca665cca478979d177de4";
+    sha256 = "1gvx7nkg4spz245hnmzcshiy0k3qvslqlzp4if7ngdqi2pfjrn4q";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "010-editor";
+  version = "12.0.1";
+
+  src = fetchurl {
+    url = "https://download.sweetscape.com/010EditorLinux64Installer${version}.tar.gz";
+    sha256 = "0ln8qxy1mij8qnnn8dxwnvmkx69l0f0w6xnb615ihx8wyhvlmfzh";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ] ++ (with perlPackages; [
+    perl
+    ModernPerl
+    CompressRawLzma
+    CompressRawZlib
+    FileHomeDir
+    TermProgressBar
+    DataDump
+  ]);
+
+  buildInputs = [
+    zlib
+    freetype
+    fontconfig
+    libxkbcommon
+    cups
+  ];
+
+  # Archive does not have a subdirectory
+  setSourceRoot = "sourceRoot=`pwd`";
+
+  # Use shipped Qt5 libs
+  runtimeDependencies = [ "$out" ];
+
+  postUnpack = ''
+    perl ${extract}/extract.pl -s Home=/ 010EditorLinux64Installer
```

Comment:
Did you build this with sandbox enabled? I would expect it to fail as soon as it tries to write something into $HOME?

Reply:
Yes (sandbox is on by default).

It doesn't actually write to `$HOME` - it writes to _install_dir_. If `Home` is not set to `/` it creates a pointless directory structure (`./install_dir/home/$USER/010editor`) instead of `./install_dir/010editor`. Other than that, it's not used.

Note that the script also supports `--prefix`, but that effectively renames `install_dir` - neither variables are in any way used to modify the installer payload.